plugin/sampling/strategystore/adaptive/processor.go

```diff
@@ -95,147 +93,134 @@ type Processor struct {
 	calculateProbabilitiesLatency metrics.Timer
+	lastCheckedTime               time.Time
 }
 
 // newProcessor creates a new sampling processor that generates sampling rates for service operations.
-func newProcessor(
+func newPostAggregator(
 	opts Options,
 	hostname string,
 	storage samplingstore.Store,
 	electionParticipant leaderelection.ElectionParticipant,
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
-) (*Processor, error) {
+) (*PostAggregator, error) {
 	if opts.CalculationInterval == 0 || opts.AggregationBuckets == 0 {
 		return nil, errNonZero
 	}
 	if opts.BucketsForCalculation < 1 {
 		return nil, errBucketsForCalculation
 	}
 	metricsFactory = metricsFactory.Namespace(metrics.NSOptions{Name: "adaptive_sampling_processor"})
-	return &Processor{
+	return &PostAggregator{
 		Options:             opts,
 		storage:             storage,
 		probabilities:       make(model.ServiceOperationProbabilities),
 		qps:                 make(model.ServiceOperationQPS),
 		hostname:            hostname,
-		strategyResponses:   make(map[string]*api_v2.SamplingStrategyResponse),
 		logger:              logger,
 		electionParticipant: electionParticipant,
 		// TODO make weightsCache and probabilityCalculator configurable
 		weightVectorCache:             NewWeightVectorCache(),
 		probabilityCalculator:         calculationstrategy.NewPercentageIncreaseCappedCalculator(1.0),
-		followerRefreshInterval:       defaultFollowerProbabilityInterval,
 		serviceCache:                  []SamplingCache{},
 		operationsCalculatedGauge:     metricsFactory.Gauge(metrics.Options{Name: "operations_calculated"}),
 		calculateProbabilitiesLatency: metricsFactory.Timer(metrics.TimerOptions{Name: "calculate_probabilities"}),
 	}, nil
 }
 
-// GetSamplingStrategy implements Thrift endpoint for retrieving sampling strategy for a service.
-func (p *Processor) GetSamplingStrategy(_ context.Context, service string) (*api_v2.SamplingStrategyResponse, error) {
-	p.RLock()
-	defer p.RUnlock()
-	if strategy, ok := p.strategyResponses[service]; ok {
+// GetSamplingStrategy implements protobuf endpoint for retrieving sampling strategy for a service.
+func (ss *StrategyStore) GetSamplingStrategy(_ context.Context, service string) (*api_v2.SamplingStrategyResponse, error) {
+	ss.RLock()
+	defer ss.RUnlock()
+	if strategy, ok := ss.strategyResponses[service]; ok {
 		return strategy, nil
 	}
-	return p.generateDefaultSamplingStrategyResponse(), nil
+	return ss.generateDefaultSamplingStrategyResponse(), nil
 }
 
-// Start initializes and starts the sampling processor which regularly calculates sampling probabilities.
-func (p *Processor) Start() error {
+// Start initializes and starts the PostAggregator which regularly calculates sampling probabilities.
+func (p *PostAggregator) Start() error {
 	p.logger.Info("starting adaptive sampling processor")
 	if err := p.electionParticipant.Start(); err != nil {
 		return err
 	}
 	p.shutdown = make(chan struct{})
```

Comment:
looks like there's nothing to start here, which makes sense since we just want the aggregator to call post-aggregator on every tick